OpenRA.Mods.Common/Traits/Player/ChangeOwner.cs

```diff
@@ -0,0 +1,21 @@
+﻿using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public abstract class ChangeOwnerInfo : ITraitInfo
+	{
+		public abstract object Create(ActorInitializer init);
+	}
+
+	public abstract class ChangeOwner
+	{
+		protected void NeedChangeOwner(Actor self, Actor actor, Player newOwner)
+		{
+			var oldOwner = self.Owner;
+			self.ChangeOwner(newOwner);
+
+			foreach (var t in self.TraitsImplementing<INotifyCapture>())
```

Comment:
Could you explain the idea behind this? This might cause issues. I can understand the idea behind it, but IMO it'd be better if you introduce a separate interface here.